container_test.go

```diff
@@ -629,6 +632,77 @@ func TestContainerKill(t *testing.T) {
 	}
 }
 
+func TestKillContainerDeletedByRunc(t *testing.T) {
+	t.Parallel()
+
+	// We skip this case when runtime is crun.
+	if os.Getenv("RUNC_FLAVOR") == "crun" {
+		t.Skip("skip it when using crun")
+	}
+
+	client, err := newClient(t, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	var (
+		image       Image
+		ctx, cancel = testContext(t)
+		id          = t.Name()
+		runcRoot    = "/tmp/runc-test"
+	)
+	defer cancel()
+
+	image, err = client.GetImage(ctx, testImage)
+	if err != nil {
+		t.Fatal(err)
+	}
+	container, err := client.NewContainer(ctx, id,
+		WithNewSnapshot(id, image),
+		WithNewSpec(oci.WithImageConfig(image), withProcessArgs("sleep", "10")),
+		WithRuntime(client.runtime, &options.Options{Root: runcRoot}))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer container.Delete(ctx)
+
+	task, err := container.NewTask(ctx, empty())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer task.Delete(ctx)
+
+	statusC, err := task.Wait(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := task.Start(ctx); err != nil {
+		t.Fatal(err)
+	}
+
+	rcmd := &runc.Runc{
+		Root: path.Join(runcRoot, testNamespace),
+	}
+
+	if err := rcmd.Delete(ctx, id, &runc.DeleteOpts{Force: true}); err != nil {
+		t.Fatal(err)
+	}
+	err = task.Kill(ctx, syscall.SIGKILL)
+	if err == nil {
+		t.Fatal("second call to kill should return an error")
+	} else if !errdefs.IsNotFound(err) {
+		t.Errorf("expected error %q but received %q", errdefs.ErrNotFound, err)
+	}
+
+	select {
+	case <-statusC:
+	case <-time.After(2 * time.Second):
+		t.Errorf("")
```

Comment:
please add `unexpected timeout when try to get exited container's status`

Reply:
Sorry, and done.